### PR TITLE
Add end to end test for SCM and file decorations

### PR DESCRIPTION
**Pull request by mattseddon**
# 2/2 `main` <- #2002 <- this

This PR adds an end to end test for the SCM view and our custom file decorations.

### CODE REVIEW

File: extension/src/test/e2e/extension.test.ts [hunk 1]
```diff
@@ -117,4 +121,48 @@ suite('DVC Extension For Visual Studio Code', () => {
...
+        {
+          interval: 5000,
+          timeout: 60000
+        }
+      )
+
+      expect(expectedScmItemLabels.sort()).toStrictEqual(
+        dvcTreeItemLabels.sort()
+      )
+    }).timeout(60000)
```
Comment (wolmir) [inline extension/src/test/e2e/extension.test.ts:166]: Isn't this timeout too long?

**Review by wolmir — APPROVED** [review]

File: extension/src/test/e2e/extension.test.ts (hunk 1 above)
Comment (mattseddon) [reply extension/src/test/e2e/extension.test.ts:166]: The SCM state relies on `dvc status`. Whenever an experiment is running `dvc status` will fail. There is an exponential back off in place for retries which means that after the experiment finishes there can be a significant lag before the SCM view is updated with the correct details. Locally this is less of a problem but in the CI running an experiment takes longer. Leading to a longer lag in the subsequent update. I'd rather that this test is reliable with a longer timeout than flaky with a shorter one. 

Does that make sense? WDYT?

Comment (wolmir) [reply extension/src/test/e2e/extension.test.ts:166]: It does for me 👍
I know this is a can of worms, but is there a way to only `dvc status`  after all the runs? Like using a subscription or something like that?

Comment (shcheklein) [reply extension/src/test/e2e/extension.test.ts:166]: Q: can it be done as a global setting - one minute timeout per test?

Comment (shcheklein) [reply extension/src/test/e2e/extension.test.ts:166]: probably it applies to all tests that run DVC internally?

Comment (mattseddon) [reply extension/src/test/e2e/extension.test.ts:166]: > I know this is a can of worms, but is there a way to only `dvc status` after all the runs? Like using a subscription or something like that?

That would mean we would have to introduce stubs into this test suite, making them less e2e. I would rather test the extension as is. We should be able to improve the situation once we get `data:status`.

> Q: can it be done as a global setting - one minute timeout per test?

This was actually already set to 1m

**Pull request merged**